### ISSUE TITLE
Support glue records

### ISSUE THREAD

**Issue by juliusrickert** [StackExchange/dnscontrol#882]
In the light of  #577, I would like to know your stance on managing glue records with DNSControl.

As I was trying to implement registrar support for a provider, I asked myself whether managing a zone with glue records with DNSControl could lead to the loss of glue records when changing the set of nameservers.

Example:
Existing nameservers:
ns1.example.com with IP address 2001:db8::1
ns2.example.com with IP address 2001:db8::2

New nameservers:
ns1.example.com
ns2.example.com
ns.example.net

If an API requires the client to re-set all nameservers when adding a new one, like [this API](https://www.hosting.de/api/#the-domain-object) does, this change would remove the existing glued addresses.

At the moment, a provider could circumvent this by remembering the IP addresses it received by the registrar.
But this could lead to inconsistencies between DNSControl and the registrar.

Therefore, I would like to extend `models.Nameserver` to be able to hold IP addresses too. But this conflicts with #577.
The `nameservers.DetermineNameservers` method could then add the IP addresses of the nameservers that are part of the zone to the respective nameserver struct.

Example:

```javascript
D('example.com', REG_EXAMPLE, DnsProvider(DNS_EXAMPLE, 0), // ignore provider's nameservers
    NAMESERVER('ns1.example.com'),
    NAMESERVER('ns2.example.com'),
    NAMESERVER('ns.example.net'),

    A('ns1', '10.0.0.1'),
    AAAA('ns1', '2001:db8::1:1'),
    A('ns2', '10.0.0.2'),
    AAAA('ns2', '2001:db8::2:1'),
    AAAA('ns2', '2001:db8::2:2')
);
```
would produce this nameserver set (for the zone example.com):
```go
nameservers := []*models.Nameserver{
	&models.Nameserver{
		Name: "ns1.example.com",
		IPs: []net.IP{
			net.IP("10.0.0.1"),
			net.IP("2001:db8::1:1"),
		},
	},
	&models.Nameserver{
		Name: "ns2.example.com",
		IPs: []net.IP{
			net.IP("10.0.0.2"),
			net.IP("2001:db8::2:1"),
			net.IP("2001:db8::2:2"),
		},
	},
	&models.Nameserver{
		Name: "ns.example.net",
	},
}
```

**Comment by tlimoncelli** [comment]
Very interesting!

Do the IP addresses need to be stored in `models.Nameserver`?  When constructing the hosting.de's Nameserver Object, couldn't you populate the `ips` field by searching the  `models.DomainConfig.Records` list?

Tom

P.S. I've been postponing https://github.com/StackExchange/dnscontrol/issues/577 in case someone finds a reason to add more fields.

**Comment by juliusrickert** [comment]
> Do the IP addresses need to be stored in `models.Nameserver`?

No, they don't.

> When constructing the hosting.de's `Nameserver` Object, couldn't you populate the `ips` field by searching the `models.DomainConfig.Records` list?

That's possible.

I just thought that it would be useful to implement this in a provider-agnostic way, as all providers capable of managing glue records would need to implement a similar mechanism.

But, to be fair, I just looked at the APIs of netcup and hosting.de, and both of them require you to do this. I now looked at some of the other providers supported by DNSControl and found that they employ different concepts of managing glue records, mostly (vanity) nameservers as a separately managed object. These could still benefit from this but not in the way netcup's or hosting.de's integration would.
